base/range.jl

```diff
@@ -47,25 +47,16 @@ function _colon(start::T, step, stop::T) where T
 end
 
 """
-    range(start[, stop]; length, stop, step=1)
+    range(start, stop; length, step)
+    range(start; length, stop, step)
+    range(;start, length, stop, step)
 
-Given a starting value, construct a range either by length or from `start` to `stop`,
-optionally with a given step (defaults to 1, a [`UnitRange`](@ref)).
-One of `length` or `stop` is required.  If `length`, `stop`, and `step` are all specified, they must agree.
-
-If `length` and `stop` are provided and `step` is not, the step size will be computed
-automatically such that there are `length` linearly spaced elements in the range.
-
-If `step` and `stop` are provided and `length` is not, the overall range length will be computed
-automatically such that the elements are `step` spaced.
-
-Special care is taken to ensure intermediate values are computed rationally.
-To avoid this induced overhead, see the [`LinRange`](@ref) constructor.
-
-`stop` may be specified as either a positional or keyword argument.
-
-!!! compat "Julia 1.1"
-    `stop` as a positional argument requires at least Julia 1.1.
+Construct a `range` from the arguments.
+Mathematically a range is uniquely determined by any three of `start`, `step`, `stop` and `length`.
+Valid invocations of range are:
+* Call `range` with any three of `start`, `step`, `stop`, `length`.
+* Call `range` with two of `start`, `stop`, `length`. In this case `step` will be assumed
+to be one. If all arguments are integers, a [`UnitRange`](@ref) will be returned.
```

Comment:
```suggestion
to be one. If both arguments are of type [`Integer`](@ref), a [`UnitRange`](@ref) will be returned.
```

> * Call `range` with two of `start`, `stop`, `length`. In this case `step` will be assumed
to be one. If all arguments are integers, a [`UnitRange`](@ref) will be returned.

```julia
julia> t = (2.0,5)
(2.0, 5)

julia> isinteger.(t)
(true, true)

julia> range_start_length(t...)
2.0:1.0:6.0

julia> range_stop_length(t...)
-2.0:1.0:2.0

julia> range_start_stop(t...)
2.0:1.0:5.0

julia> range_start_length(t...) isa UnitRange
false

julia> range_stop_length(t...) isa UnitRange
false

julia> range_start_stop(t...) isa UnitRange
false

julia> range_start_stop(BigInt.(t)...) isa UnitRange
true

julia> range_start_length(BigInt.(t)...) isa UnitRange
true

julia> range_stop_length(BigInt.(t)...) isa UnitRange
true

julia> range_start_stop(UInt8.(t)...) isa UnitRange
true

julia> range_start_length(UInt8.(t)...) isa UnitRange
true

julia> range_stop_length(UInt8.(t)...) isa UnitRange
true
```

Reply:
Just some documentation editing. Sorry to be that guy.

If we eventually use `Base.OneTo` we might want to say `AbstractUnitRange`, but that's for another PR.

Reply:
I think it is clear that the `UnitRange` comment does not apply to the first bullet point at the moment. I'll just document that here in case someone looks this up later.

```julia
julia> t = (1,1,3)
(1, 1, 3)

julia> isa.(t,Integer)
(true, true, true)

julia> range_start_step_length(t...)
1:1:3

julia> range_start_step_length(t...) isa UnitRange
false
```

If we really wanted this to produce a UnitRange, then maybe should consider using `Val` for `step`. I'm not sure if that is worth it.

Reply:
I would lean towards the simpler: `If all arguments are Integers`.  No need to overly cross-ref — just capitalize it.